### ISSUE TITLE
403 Error [URGENT]

### ISSUE THREAD

**Issue by Dlas1212** [hungdev/react-native-instagram-login#102]
Calling 'https://api.instagram.com/oauth/access_token' returning Forbidden 403. Started yesterday randomly.

I am on the latest version too.

**Comment by unixkapl** [comment]
+1

**Comment by alyn3d** [comment]
+1

**Comment by saurabhjayx** [comment]
same happening with me

**Comment by alyn3d** [comment]
As far as I can tell there's an issue with the **redirectUrl**
Still haven't figured how to solve it.

**Comment by Tymofiev** [comment]
What I noticed is that when it's first time I logged into IG then it worked just fine, and if it's second attempt so browser kinda had already remembered my account - it failed with 403.
![telegram-cloud-photo-size-2-5336794829092403584-y](https://user-images.githubusercontent.com/44410158/146921411-d27183fe-9001-401d-b74a-154205996e1a.jpg)

I suppose it is the cookies issue, as it returned html page with the text that said smth about `cookies` couple times.

**Comment by hungdev** [comment]
> What I noticed is that when it's first time I logged into IG then it worked just fine, and if it's second attempt so browser kinda had already remembered my account - it failed with 403.
> 
> I suppose it is the cookies issue, as it returned html page with the text that said smth about `cookies` couple times.

I don't think so, because I tried with props `incognito` but it still gets that issue.

**Comment by sujay-bidchat** [comment]
+1

**Comment by sujay-bidchat** [comment]
Adding to all the above comments. This issue comes only for Android. It works perfectly fine for iOS every time. Even it works every time on Postman too.

**Comment by nimit07** [comment]
@sujay-bidchat  were you able to find a solution for this?

**Comment by sujay-bidchat** [comment]
@nimit07 I have found a workaround for this issue. You can send the "code" to BE and let them get the access_token for you. This is what I've done.

> @sujay-bidchat were you able to find a solution for this?

**Comment by saurabh-cimpress** [comment]
Having the same issue. Any workaround or Fix?

**Comment by alyn3d** [comment]
Workaround that I ended up using is [this](https://github.com/hungdev/react-native-instagram-login/issues/80#issuecomment-998026216)

**Comment by sujay-bidchat** [comment]
> Workaround that I ended up using is [this](https://github.com/hungdev/react-native-instagram-login/issues/80#issuecomment-998026216)

I have done the same as mentioned above.

**Comment by duongquang1611** [comment]
does anyone have a solution?

**Comment by pnishith** [comment]
Try in a this way:

`
const url = 'oauth/access_token';
const api = axios.create({
  baseURL: 'https://api.instagram.com/',
  withCredentials: false, // Disable automatic handling of cookies
});
api
    .post(url, new URLSearchParams(data).toString(), config)
    .then((response) => {
      console.log(response);
      // Handle the response containing the access token
      const accessToken = response.data.access_token;
      // Use the access token to make further API requests
      // ...
    })
    .catch((error) => {
      console.log(error);
      // Handle any error that occurred during the request
    });
`
